io/src/pcd_io.cpp

```diff
@@ -1427,21 +1447,22 @@ pcl::PCDWriter::writeBinaryCompressed (std::ostream &os, const pcl::PCLPointClou
   }
 
   std::vector<char> temp_buf (data_size * 3 / 2 + 8);
-  // Compress the valid data
-  unsigned int compressed_size = pcl::lzfCompress (&only_valid_data.front (), 
-                                                   static_cast<unsigned int> (data_size), 
-                                                   &temp_buf[8], 
-                                                   data_size * 3 / 2);
-  // Was the compression successful?
-  if (compressed_size == 0)
-  {
-    return (-1);
+  if (data_size != 0) {
```

Comment:
I think it would also be nice here to explicitly set all 8 chars/bytes of `temp_buf` to zero if `data_size` is zero (to indicate that `compressed_size` and `data_size` are zero)